File: mdd-remote-dtos/src/main/java/io/mateu/remote/dtos/Crud.java
package io.mateu.remote.dtos;

import lombok.*;

import java.util.List;
import java.util.Map;

@Data@Builder
@NoArgsConstructor(access = AccessLevel.PACKAGE)@AllArgsConstructor(access = AccessLevel.PACKAGE)
public class Crud implements ViewMetadata {

    private final ViewMetadataType type = ViewMetadataType.Crud;

    private String dataPrefix;

    private String listId;

    private String title;

    private String subtitle;

    private boolean canEdit;

    private SearchForm searchForm;

    private List<Column> columns;

    private List<Action> actions;

}

File: frontend/src/mateu/shared/apiClients/dtos/Crud.ts
import ComponentMetadata from "./ComponentMetadata";
import SearchForm from "./SearchForm";
import Column from "./Column";
import Action from "./Action";

export default interface Crud extends ComponentMetadata {

    title: string
    subtitle: string
    canEdit: boolean
    searchable: boolean
    searchForm: SearchForm
    columns: Column[]
    actions: Action[]
    child: boolean

}
